locals.tf

```diff
@@ -8,6 +8,7 @@ locals {
   cluster_security_group_id = var.cluster_security_group_id == null ? aws_security_group.cluster_security_group.0.id : var.cluster_security_group_id
   workers_security_group_id = var.workers_security_group_id == null ? aws_security_group.workers_security_group.0.id : var.workers_security_group_id
   cluster_name              = "${var.prefix}-eks"
+  tags                      = length(var.tags) == 0 ? { project_name = "viya" } : var.tags
```

Comment:
Need to also check for null. A user could do this as well. If the user sets the value to null I believe your length check will fail. Code would look like
```terraform
tags = var.tags == null ? { project_name = "viya" } : length(var.tags) == 0 ? { project_name = "viya" } : var.tags
```
you might also thing about having a default tag local so that if we need to change what the default is, it's not a hunt and paste for `{ project_name = "viya" }` throughout the code.

Reply:
Added null comparison, the length check fails with a null entry as you indicated. Also added a local default_tags value to avoid literal duplication in locals.